agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/reminder/ReminderDetailScreen.kt

```diff
@@ -87,106 +86,59 @@ fun ReminderDetailScreenRoot(
 internal fun ReminderDetailScreen(
     state: ReminderDetailState,
     onAction: (ReminderDetailAction) -> Unit,
-    clock: Clock = Clock.System,
 ) {
     val spacing = LocalSpacing.current
+    val clock = LocalClock.current
     val dividerColor = TaskyWhite2
 
-    TaskyScaffold(
-        topAppBar = {
-            AgendaDetailToolbar(
-                title = if (state.isEditing) {
-                    stringResource(id = R.string.edit_reminder)
-                } else {
-                    state.selectedDate.formatFullDate(clock).asString()
-                },
-                onCloseClick = { onAction(ReminderDetailAction.OnClose) },
-                isEditing = state.isEditing,
-                onEditClick = { onAction(ReminderDetailAction.OnEnableEdit) },
-                onSaveClick = { onAction(ReminderDetailAction.OnSave) },
-            )
-        },
-    ) { innerPadding ->
-        Column(
-            modifier = Modifier
-                .padding(innerPadding)
-                .fillMaxSize()
-                .verticalScroll(rememberScrollState())
-                .clip(
-                    RoundedCornerShape(
-                        topStart = spacing.scaffoldContainerRadius,
-                        topEnd = spacing.scaffoldContainerRadius,
-                    ),
-                )
-                .background(
-                    MaterialTheme.colorScheme.background,
-                    shape = RoundedCornerShape(
-                        topStart = spacing.scaffoldContainerRadius,
-                        topEnd = spacing.scaffoldContainerRadius,
-                    ),
+    if (state.isEditingTitle) {
+        AgendaDetailEditTextScreen(
+            type = EditTextType.Title,
+            value = state.title,
+            onBackClick = {
+                onAction(ReminderDetailAction.OnCloseTitle)
+            },
+            onSaveClick = { newTitle ->
+                onAction(ReminderDetailAction.OnSaveTitle(newTitle))
+            },
+        )
+    } else if (state.isEditingDescription) {
+        AgendaDetailEditTextScreen(
+            type = EditTextType.Description,
+            value = state.description ?: "",
+            onBackClick = {
+                onAction(ReminderDetailAction.OnCloseDescription)
+            },
+            onSaveClick = { newDescription ->
+                onAction(ReminderDetailAction.OnSaveDescription(newDescription))
+            },
+        )
+    } else {
```

Comment:
Why did you decide for not making the edit screens separate screens (in the end it is its own screen, right?)

I see that it's easier to share the state here, but this comes with quite some side-effects people are often not aware of. For example, back navigation will break by default since the edit screen is not on its own back stack entry. You'd need to handle that. Also if you need deep linking at some point to these edit or photo screens, that doesn't work without workarounds.

Reply:
ah, good point with the workarounds needed to support deeplinking, hadn't thought of that... I knew I was gonna override the backhandler anyways on the edit text screen so that I could show a cancel/confirm dialog for the back press button. And I thought it would make sense to try to keep the nav graph less cluttered.

To do this with them being their own screens and being navigable by compose navigation, I'd need to use nav args for both the detail screen and the edit text screen, right? So that I could send the updated title or description back to the detail screen to save?

Reply:
Perhaps NavigableListDetailScaffold would help me out here?

Reply:
The scaffold comes with the same problems, since it doesn't work with the real nav controller. For Tasky it would be fine as we don't need deep linking to these detail screens, but these are things that could change in the real world of course. 
To pass back arguments from one screen to another, you can use `navController.previousBackStackEntry.savedStateHandle` to put some values in there and access them on the previous screen.